signalfx/resource_signalfx_slo_chart.go

```diff
@@ -0,0 +1,122 @@
+// Copyright Splunk, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package signalfx
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	chart "github.com/signalfx/signalfx-go/chart"
+)
+
+func sloChartResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"slo_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of the attached SLO",
+			},
+			"url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "URL of the chart",
+			},
+		},
+
+		Create: slochartCreate,
+		Read:   slochartRead,
+		Update: slochartUpdate,
+		Delete: slochartDelete,
+		Exists: chartExists,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+/*
+Use Resource object to construct json payload in order to create an SLO chart
+*/
+func getPayloadSloChart(d *schema.ResourceData) *chart.CreateUpdateSloChartRequest {
+	return &chart.CreateUpdateSloChartRequest{
+		SloId: d.Get("slo_id").(string),
+	}
+}
+
+func slochartCreate(d *schema.ResourceData, meta any) error {
+	config := meta.(*signalfxConfig)
+	payload := getPayloadSloChart(d)
+
+	debugOutput, _ := json.Marshal(payload)
+	log.Printf("[DEBUG] SignalFx: Create SLO Chart Payload: %s", string(debugOutput))
```

Comment:
For this, you could do:

```go
tflog.Debug(ctx, "Created SLO chart", tfext.NewLogFields().JSON("payload", payload))
```

This would keep the code more concise.